modules/puppet_proxy/puppet_api.rb

```diff
@@ -7,13 +7,9 @@ class Proxy::Puppet::Api < ::Sinatra::Base
 
   inject_attr :class_retriever_impl, :class_retriever
   inject_attr :environment_retriever_impl, :environment_retriever
-  inject_attr :puppet_runner_impl, :puppet_runner
 
   post "/run" do
-    log_halt 400, "Failed puppet run: No nodes defined" unless params[:nodes]
-    log_halt 500, "Failed puppet run: Check Log files" unless puppet_runner.run([params[:nodes]].flatten)
-  rescue => e
-    log_halt 500, "Failed puppet run: #{e}"
+    raise NotImplementedError.new
```

Comment:
Isn't this just a regular exception? Does it return a HTTP 500 or 501? Perhaps a test would be nice. Also a message that puppetrun support has been removed in version 2.3. is nice for the consumer for the API.

Reply:
@ekohl done.  thanks!